tests/upgrade.bats

```diff
@@ -5,70 +5,54 @@ setup() {
-	wait_for_cluster_admission_policy PolicyUniquelyReachable
+function helm_in {
+    helm upgrade --install --wait --namespace $NAMESPACE --create-namespace \
+        "${@:2}" $1 $KUBEWARDEN_CHARTS_LOCATION/$1
 }
 
-@test "[CRD upgrade] Launch a privileged pod should fail" {
-	kubectl_apply_should_fail $RESOURCES_DIR/violate-privileged-pod-policy.yaml
+# check_apiversion admissionpolicies v1
+function check_apiversion {
+    run -0 bash -c "kubectl get $1 -o json | jq -er '.items[].apiVersion' | uniq"
+    assert_output "policies.kubewarden.io/$2"
 }
 
-@test "[CRD upgrade] Upgrade CRDs" {
-	helm upgrade --wait \
-		--namespace $NAMESPACE --create-namespace \
-		--version $KUBEWARDEN_CRDS_CHART_VERSION \
-		kubewarden-crds $KUBEWARDEN_CHARTS_LOCATION/kubewarden-crds
+function check_default_policies {
+    # Check all policies are in v1
+    check_apiversion clusteradmissionpolicies v1
+    wait_for_cluster_admission_policy PolicyUniquelyReachable
 
-	helm upgrade --wait --namespace $NAMESPACE \
-		--values $RESOURCES_DIR/default-kubewarden-controller-values.yaml \
-		--version $KUBEWARDEN_CONTROLLER_CHART_VERSION \
-		kubewarden-controller $KUBEWARDEN_CHARTS_LOCATION/kubewarden-controller
+    # Run privileged pod (should fail)
+    # kubefail_privileged run pod-privileged --image=k8s.gcr.io/pause --privileged
+    # Workaround - https://suse.slack.com/archives/C02DBSK7HC1/p1661518752112929
+    run -1 kubectl run pod-privileged --image=k8s.gcr.io/pause --privileged
+}
 
-	yq --in-place -Y ".recommendedPolicies.enabled=true" $RESOURCES_DIR/default-kubewarden-defaults-values.yaml
-	yq --in-place -Y ".recommendedPolicies.defaultPolicyMode=\"protect\"" $RESOURCES_DIR/default-kubewarden-defaults-values.yaml
-	helm upgrade --wait --namespace $NAMESPACE \
-		--values $RESOURCES_DIR/default-kubewarden-defaults-values.yaml \
-		--version $KUBEWARDEN_DEFAULTS_CHART_VERSION \
-		kubewarden-defaults $KUBEWARDEN_CHARTS_LOCATION/kubewarden-defaults
 
-	crd_version=$(kubectl --context $CLUSTER_CONTEXT get clusteradmissionpolicy -o json | jq -r ".items[].apiVersion" | uniq)
-	[ "$crd_version" = "policies.kubewarden.io/v1" ]
-	wait_for_cluster_admission_policy PolicyUniquelyReachable
+@test "[CRD upgrade] Install old Kubewarden" {
+    # Install old kubewarden version
+    helm_in kubewarden-crds --version $KUBEWARDEN_CRDS_CHART_OLD_VERSION
+    helm_in kubewarden-controller --version $KUBEWARDEN_CONTROLLER_CHART_OLD_VERSION
+    helm_in kubewarden-defaults --version $KUBEWARDEN_DEFAULTS_CHART_OLD_VERSION \
+        --set recommendedPolicies.enabled=True \
+        --set recommendedPolicies.defaultPolicyMode=protect
+    check_default_policies
 }
 
-@test "[CRD upgrade] Launch a privileged pod should fail after CRD upgrade" {
-	kubectl_apply_should_fail $RESOURCES_DIR/violate-privileged-pod-policy.yaml
+@test "[CRD upgrade] Upgrade Kubewarden" {
+    helm_in kubewarden-crds --version $KUBEWARDEN_CRDS_CHART_VERSION
```

Comment:
I expect in the future to have 2 jobs:

A. Upgrade from "Generally Available Version" to "Version Under Test", where:
   * `$KUBEWARDEN_FOO_CHART_OLD_VERSION` matches what is released on charts.kubewarden.io.
   * `$KUBEWARDEN_FOO_CHART_VERSION` is the version under test in the PR, nightly from main, etc.
   
B. Upgrade from "Version Under Test" to `Version Under Test +1`, to test that we aren't breaking upgradabilty, and if we are introducing any bugs, we have the capability to fix them by issuing an upgrade instead of instructing users to reinstall from scratch. Where:
   * `$KUBEWARDEN_FOO_CHART_OLD_VERSION` is the version under test in the PR, nightly from main, etc.
   * `$KUBEWARDEN_FOO_CHART_VERSION` is a newly created version just for the test, fabricated by bumping the `version`s of the charts, and possibly changing their default configs.

Reply:
makes sense, thank you